### ISSUE TITLE
Problem with train_batch bounding boxes

### ISSUE THREAD

**Issue by balogistvan** [ultralytics/ultralytics#16404]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello. I have a problem, and I dont know, how solve it. When I train yolov8, with costom dataset wich I make in roboflow, in train bathes have bad and false bounding boxes.
If you can help me solve this problem, thank you.
![train_batch1 (1)](https://github.com/user-attachments/assets/12c56945-d707-49bf-93ec-9a7fb30692fe)
![train_batch2](https://github.com/user-attachments/assets/133e9229-33df-46b3-8ebf-5f04aca9ec53)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @balogistvan, thank you for reaching out and using Ultralytics 🚀!

We recommend new users visit the [Docs](https://docs.ultralytics.com) where there are many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples. These resources may help answer some common questions.

If this is a 🐛 Bug Report, could you please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug the issue effectively?

For custom training ❓, please include as much detail as possible, such as dataset image examples, training logs, and ensure you’re following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join our Ultralytics community where you feel most comfortable:
- Real-time chat on [Discord](https://ultralytics.com/discord) 🎧
- In-depth discussions on [Discourse](https://community.ultralytics.com)
- Share and learn on our [Subreddit](https://reddit.com/r/ultralytics)

## Upgrade

Make sure you’re using the latest `ultralytics` package along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLOv8 is supported in various environments:
- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon**: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

This badge indicates the status of Ultralytics CI tests, which validate operation across different systems.

This is an automated response, but an Ultralytics engineer will assist you soon.

**Comment by Y-T-G** [comment]
What's the training code you're using?

**Comment by balogistvan** [comment]
I use code in colab.
This code: https://colab.research.google.com/github/roboflow/notebooks/blob/main/notebooks/train-yolov8-object-detection-on-custom-dataset.ipynb

**Comment by Y-T-G** [comment]
That's a really old ultralytics version. You should follow the official ultralytics guide.

https://docs.ultralytics.com/modes/train/

**Comment by balogistvan** [comment]
Thanks, but the problem still exists. Could it be that the dataset has both a bounding box and a polygon?

**Comment by Y-T-G** [comment]
Probably.

**Comment by balogistvan** [comment]
Thanks for your help. The problem is solved. The problem was the polygons and bounding boxes in same dataset. When I deleted all polygons, the batch is correct.

**Comment by glenn-jocher** [comment]
You're welcome! Glad to hear that resolved the issue. If you have any more questions, feel free to ask.
